### DISCUSSION TITLE
No audio with WebRTC

### DISCUSSION THREAD

**Discussion by AxelHamburch** [bluenviron/mediamtx#2824] [Questions]
### Question

My server recodes a stream after this format.

ffmpeg -i rtsp://original-source \
-pix_fmt yuv420p -c:v libx264 -preset ultrafast -b:v 600k \
-c:a aac -b:a 160k \
-f rtsp rtsp://localhost:8554/mystream

The source is an RTSP stream with audio, or I have also tried a simple video format with audio.

When I call the stream with a browser in WebRTC format "http: // localhost: 8889/mystream", then I get the video stream, but no audio signal. No matter which browser or which source I use.

Is that not possible with WebRTC or what do I do wrong?

**Comment by aler9** [comment]
Hello, as clearly written at the beginning of the README, WebRTC doesn't support AAC. This is a limitation of the current specification and is strictly followed by all browsers. Use Opus.

**Reply by AxelHamburch** [reply]
Thank you!

Is it that simple like this?

ffmpeg -i rtsp://original-source
-pix_fmt yuv420p -c:v libx264 -preset ultrafast -b:v 600k
-c:a libopus -b:a 160k
-f rtsp rtsp://localhost:8554/mystream

Or do I have a mistake?

**Comment by AxelHamburch** [comment]
Thanks, it finally worked with "libopus". 🙏

**Reply by AseCoder** [reply]
Hi, what ffmpeg command did you end up using? I have a problem where the audio becomes distorted when I re-encode using libopus.

**Reply by AxelHamburch** [reply]
Hi there, my command is a little more complicated because it picks up a stream and displays a QR code at the bottom right. On top there is an audio mp3 file that plays the music in a loop. But I'm happy to share it.

ffmpeg -rtsp_transport tcp -stream_loop -1 -i "rtsp://myone.duckdns.org/mjpeg/1" \
-i qr128.png -stream_loop -1 -i music.mp3 -filter_complex "[0:v][1:v]overlay=W-w-5:H-h-5[v];\
[2:a]aloop=loop=-1[outa]" -map "[v]" -map "[outa]" -pix_fmt yuv420p -r:v 30 -c:v libx264 \
-preset ultrafast -b:v 600k -c:a libopus -b:a 128k -f rtsp rtsp://localhost:8554/mystream

The stream can be read in the web browser with "http://<MyServerIP>:8889/mystream/".
It works so far. However, every now and then I have short interruptions in the sound. No idea what it is, probably the transmission as there is hardly any buffer in it. Maybe someone has an idea what it is.

**Reply by AseCoder** [reply]
Thanks. Sadly this didn't fix my issue #2787. Your remark about interruptions in the sound is interesting because it could be the cause for my issue. In my case, though, the interruptions are not intermittent, but the audio is constantly distorted.

**Reply by AxelHamburch** [reply]
You can check the sound here:

https://rtsp.ereignishorizont.xyz/mystream/

It takes few seconds but afters a while you will realize short interruptions. Don't know here it comes from. 🤷‍♂️

[a comment by an automated account is not shown]
